src/resource.ts

```diff
@@ -118,7 +162,17 @@ export class EnvResourceDetector {
     return true;
   }
 
+  /**
+   * Determines whether the given String is a valid printable ASCII string with
+   * a length greater than 0 and not exceed _MAX_LENGTH characters.
+   *
+   * @param str The String to be validated.
+   * @returns Whether the String is valid and not empty.
+   */
   private _isValidAndNotEmpty(str: string): boolean {
     return str.length > 0 && this._isValid(str);
   }
 }
+
+export const envDetector = new EnvDetector();
```

Comment:
Is this used anywhere?

Reply:
Nope, not yet. Straight from upstream.
Makes sense on the other hand because it has no internal state.

Reply:
I'll merge for now, we can continue the discussion.